src/SIS_sponge.F90

```diff
@@ -0,0 +1,991 @@
+  integer :: i, j, k, is, ie, js, je, ncat
+  integer :: isd, ied, jsd, jed
+  integer :: isc, iec, jsc, jec
+  integer :: itestG, jtestG, itest, jtest  ! Test point indices global and local
+  integer :: year     !< The current model year
+  integer :: day      !< The current model year-day
+  integer :: second   !< The second of the day
+  integer :: mon, hr, minute, itick
+  integer :: start_of_day, num_days
+  integer :: verbosity   !< MOM verbosity level
+  real :: max_rlxrate, rho_ice
+  integer, dimension(4) :: siz
+  character(len=40) :: ithck_var, iarea_var, rlxrate_var, rlx_unit
+  character(len=40) :: mdl = "initialize_icerelax_file"
+  character(len=50) :: rlx_long_name
+  character(len=200) :: relaxrate_file, state_file  ! relax filenames: inverse time, target fields
+  character(len=200) :: filename, inputdir ! Strings for file/path and path.
+  character(len=256) :: mesg
+
+  isc = G%isc ; iec = G%iec ; jsc = G%jsc ; jec = G%jec ; ncat = IG%CatIce
+  isd = G%isd ; ied = G%ied ; jsd = G%jsd ; jed = G%jed
+
+  Irelax = 0.0 ; itestG = 0 ; jtestG = 0 ; itest = 0 ; jtest = 0
+
+  verbosity = MOM_get_verbosity()
+
+  call get_param(param_file, mdl, "INPUTDIR", inputdir, default=".")
+  inputdir = slasher(inputdir)
+  call get_param(param_file, mdl, "ISPONGE_RELAX_FILE", relaxrate_file, &
+                 "The name of the file with the sponge relaxation rates.", &
+                 fail_if_missing=.true.)
+  call get_param(param_file, mdl, "ISPONGE_STATE_FILE", state_file, &
+                 "The name of the file with the state to relax toward.", &
+                 fail_if_missing=.true.)
+  call get_param(param_file, mdl, "ISPONGE_ITHCK_VAR", ithck_var, &
+                 "The name of the ice thickness variable in "//&
+                 "ISPONGE_STATE_FILE.", default="ithkn")
+  call get_param(param_file, mdl, "ISPONGE_IAREA_VAR", iarea_var, &
+                 "The name of the ice partial area/concentration variable in "//&
+                 "ISPONGE_STATE_FILE.", default="ithkn")
+  call get_param(param_file, mdl, "ISPONGE_RLXRATE_VAR", rlxrate_var, &
+                 "The name of the relaxation rate variable in "//&
+                 "ISPONGE_RELAX_FILE.", default="relax_rate")
+  call get_param(param_file, mdl, "ISPONGE_ITEST", itestG, &
+                 "I index of a test point to check ice relaxation dumped to log file")
+  call get_param(param_file, mdl, "ISPONGE_JTEST", jtestG, &
+                 "J index of a test point to check ice relaxation dumped to log file")
+
+  ! Read in relaxation rate, s-1, for ice thickness and partial area
+  filename = trim(inputdir)//trim(relaxrate_file)
+  call log_param(param_file, mdl, "INPUTDIR/ISPONGE_RELAX_FILE", filename)
+  if (.not.file_exists(filename, G%Domain)) &
+    call SIS_error(FATAL, " initialize_icerelax_file: Unable to open "//trim(filename))
+
+  call MOM_read_data(filename, rlxrate_var, Irelax(:,:), G%Domain, scale=US%s_to_T)
+  max_rlxrate =  maxval(Irelax*US%T_to_s)
```

Comment:
Please move lines 176 and 177 setting `max_rlxrate` and the time variables into the `if (verb_msg > verbosity)` block, as they are not used otherwise.